contrib/scalar/scalar.c

```diff
@@ -1440,6 +1456,24 @@ static int cmd_run(int argc, const char **argv)
 	return 0;
 }
 
+static int remove_deleted_enlistment(struct strbuf *path)
+{
+	int res = 0;
+	strbuf_realpath_forgiving(path, path->buf, 1);
+
+	if (run_git("config", "--global",
+			"--unset", "--fixed-value",
+			"scalar.repo", path->buf, NULL) < 0)
+		res = -1 || res;
```

Comment:
I think this `|| res` will be ignored because of short-circuiting the OR (the `-1` already satisfies it).

Reply:
Thanks! Silly mistake on my part 🤦